recipes/fmt/all/conanfile.py

```diff
@@ -71,44 +63,48 @@ def package_id(self):
             del self.info.options.with_fmt_alias
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
-
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["FMT_DOC"] = False
-        self._cmake.definitions["FMT_TEST"] = False
-        self._cmake.definitions["FMT_INSTALL"] = True
-        self._cmake.definitions["FMT_LIB_DIR"] = "lib"
+        tools.get(**self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
+
+    def layout(self):
+        cmake_layout(self)
+
+    def generate(self):
+        cmake_toolchain = CMakeToolchain(self)
+
+        cmake_toolchain.variables["FMT_DOC"] = False
+        cmake_toolchain.variables["FMT_TEST"] = False
+        cmake_toolchain.variables["FMT_INSTALL"] = True
+        cmake_toolchain.variables["FMT_LIB_DIR"] = "lib"
+
         if self._has_with_os_api_option:
-            self._cmake.definitions["FMT_OS"] = self.options.with_os_api
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
+            cmake_toolchain.variables["FMT_OS"] = self.options.with_os_api
+
+        cmake_toolchain.generate()
```

Comment:
I am not familiar with this new mechanism, but how the CMake "out of source" build work? I would assume its opt-in since not all projects support it but I don't see anything obvious to configure it 🤔